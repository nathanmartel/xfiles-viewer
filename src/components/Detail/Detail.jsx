import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';

const Detail = () => {
  
  const [charname, setName] = useState('Lorem Ipsum');
  const [image, setImage] = useState('https://placekitten.com/200/200');
  const [description, setDescription] = useState('Lorem ipsum dolor sit amet!');

  const { name } = useParams();


  useEffect(() => {
    fetch(`https://xfiles-api.herokuapp.com/api/v1/characters/${name}`)
      .then(res => res.json())
      .then(data => {
        console.log('data is', data);
        setName(data[0].name);
        setImage(data[0].image);
        setDescription(data[0].description);
      });
  }, []);

  return (
    <>
      <p>Detail</p>
      <img src={image} alt={charname} />
      <h3>{charname}</h3>
      <p>{description}</p>
    </>
  );
};

// Detail.propTypes = {
//   image: PropTypes.string.isRequired,
//   name: PropTypes.string.isRequired,
//   description: PropTypes.string.isRequired
// };

export default Detail;
