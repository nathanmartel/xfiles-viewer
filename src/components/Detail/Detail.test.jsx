import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail';

describe('Detail component', () => {
  it('renders Detail', () => {
    const monster = {
      name: 'Lorem Ipsum',
      image: 'https://placekitten.com/200/200',
      description: 'Dolor sit amet!'
    };
    const wrapper = shallow(<Detail {...monster} />);
    expect(wrapper).toMatchSnapshot();
  });
});
