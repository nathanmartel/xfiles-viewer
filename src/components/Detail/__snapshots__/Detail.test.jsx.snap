// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component renders Detail 1`] = `
<Fragment>
  <img
    alt="Lorem Ipsum"
    src="https://placekitten.com/200/200"
  />
  <h3>
    Lorem Ipsum
  </h3>
  <p>
    Dolor sit amet!
  </p>
</Fragment>
`;
